### PR TITLE
feat: add CloudFront to QR code site

### PR DESCRIPTION
**Pull request by patheard**
# Summary
This creates a CloudFront distribution for QR code that does the following:

1. Cache all GET/HEAD requests, respecting response cache headers.
2. Cache `/static/*` GET/HEAD requests for 1 week, relying on cache busting in asset filenames to keep cached items fresh.

# Expected changes
1. Create QR code CloudFront distribution.
2. Create S3 logging bucket for CloudFront.
2. Create Kinesis delivery stream for QR code WAF logs.
2. Update QR code DNS record to point to CloudFront.
3. Create new WAF ACL for QR code load balancers (only allows requests with custom HTTP header from CloudFront).
4. Attach existing QR code WAF ACL to CloudFront.
5. Update MaintenanceMode CloudFront distribution to a lower price class (since we only allow traffic from Canada).

The new traffic flow will now be:
```
Request → CloudFront → ALB → ECS
```

Related #682

### CODE REVIEW

**Comment by patheard** [comment]
TF security failure is being caused by the tfsec action failing.  Will fix in a separate PR:
```
go: downloading github.com/tfsec/tfsec v0.45.8
go get: github.com/tfsec/tfsec@v0.45.4 updating to
	github.com/tfsec/tfsec@v0.45.8: parsing go.mod:
	module declares its path as: github.com/aquasecurity/tfsec
	        but was required as: github.com/tfsec/tfsec
/entrypoint.sh: line 20: /go/bin/tfsec: No such file or directory
```

File: config/terraform/aws/s3.tf [hunk 1]
```diff
@@ -98,3 +98,44 @@ resource "aws_s3_bucket_policy" "web_distribution" {
...
+  }
+
+  # awslogsdelivery account needs full control for cloudfront logging
+  # https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/AccessLogs.html#AccessLogsBucketAndFileOwnership
+  grant {
+    id          = "c4c1ede66af53448b93c283ce9448c4ba468c9432aa01d700d3878632f77d2d0"
+    type        = "CanonicalUser"
+    permissions = ["FULL_CONTROL"]
+  }
+}
```
[the remark this answers is not in the document]
Comment (patheard) [reply config/terraform/aws/s3.tf]: This can be suppressed in Staging.

File: config/terraform/aws/cloudfront.tf [hunk 2]
```diff
@@ -95,10 +95,10 @@ resource "aws_cloudfront_distribution" "qrcode" {
 
   aliases = ["register.covid-hcportal.cdssandbox.xyz"]
 
-  # Cache GET/HEAD respecting cache-control headers (min_ttl = 0)
+  # By default, cache nothing
   default_cache_behavior {
     allowed_methods  = ["GET", "HEAD", "OPTIONS", "POST"]
-    cached_methods   = ["GET", "HEAD"]
+    cached_methods   = []
```
Comment (dorianjp) [inline config/terraform/aws/cloudfront.tf:101]: thanks @patheard

**Review by dorianjp — APPROVED** [review]
lgtm

**Pull request merged**